src/plugins/home/server/services/sample_data/routes/list.ts

```diff
@@ -105,19 +112,25 @@ export const createListRoute = (router: IRouter, sampleDatasets: SampleDatasetSc
             return;
           }
         }
-        try {
-          await context.core.savedObjects.client.get('dashboard', sampleDataset.overviewDashboard);
-        } catch (err) {
-          if (context.core.savedObjects.client.errors.isNotFoundError(err)) {
-            sampleDataset.status = NOT_INSTALLED;
+
+        // check dashboards only if a default dashboard is set
+        if (sampleDataset.overviewDashboard) {
```

Comment:
The `overviewDashboard` isn't the same as original sample data set. It's already been overridden by `sampleDataset.getDataSourceIntegratedDashboard`. It will be a positive value when MDS or workspace enabled. Maybe we need to add check logic in [line 64](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8587/files#diff-8b8661bac0ffa6640923004bae947f354ed273962a3796aabbf489dcde3a10b8L64-R67).

Reply:
Yes, updated the the `list.ts` file to check for  `sampleDataset.overviewDashboard` before using `sampleDataset.getDataSourceIntegratedDashboard`

Reply:
Good Catch, thanks!